### PR TITLE
Added details fragment using safe args

### PR DESCRIPTION
**Pull request by Obehi234**
Used safe args to pass pokemon-specific arguments to new Fragment and to make subsequent network calls.

PR_directory

PR_details

### CODE REVIEW

File: app/src/main/java/com/example/pokedex/api/ApiService.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import com.example.pokedex.pokemonDetails.detailsModel.DetailsBluePrint
+import com.example.pokedex.pokemonDirectory.pokemonModel.PokemonList
+import retrofit2.Response
+import retrofit2.http.GET
+import retrofit2.http.Path
+
+interface ApiService {
+
+    @GET("pokemon/")
+    suspend fun getPokemonForPokedex() : Response<PokemonList>
```
Comment (IfyUnoka4) [inline app/src/main/java/com/example/pokedex/api/ApiService.kt]: Rename this function

File: app/src/main/java/com/example/pokedex/api/RetrofitInstance.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+object RetrofitInstance {
+
+    private const val BASE_URL = "https://pokeapi.co/api/v2/"
```
Comment (IfyUnoka4) [inline app/src/main/java/com/example/pokedex/api/RetrofitInstance.kt]: create a constants file and move this BASE_URL to it

File: app/src/main/java/com/example/pokedex/pokemonDetails/PokemonDetailsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+class PokemonDetailsFragment : Fragment() {
+    private val args: PokemonDetailsFragmentArgs by navArgs()
+    private var _binding: FragmentPokemonDetailsBinding? = null
+    private val binding get() = _binding!!
+    private lateinit var tabLayout: TabLayout
+    private lateinit var viewPager: ViewPager2
+    private lateinit var imageView: ImageView
+    private lateinit var nameTv: TextView
+    private lateinit var numberTv: TextView
```
Comment (IfyUnoka4) [inline app/src/main/java/com/example/pokedex/pokemonDetails/PokemonDetailsFragment.kt]: you do not need these late init initialisations when using binding. Use the binding directly

File: app/src/main/java/com/example/pokedex/pokemonDetails/PokemonDetailsFragment.kt [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        detailsViewModel = ViewModelProvider(this)[DetailsViewModel::class.java]
+        pokemonName = args.pokemonName
+        pokemonNumber = args.pokemonNumber
+
+        _binding = FragmentPokemonDetailsBinding.inflate(inflater, container, false)
+        nameTv = binding.nameTV
+        nameTv.text = pokemonName
```
Comment (IfyUnoka4) [inline app/src/main/java/com/example/pokedex/pokemonDetails/PokemonDetailsFragment.kt]: you can have 
`binding.apply { 
nameTV = pokemon name
numberTV = formattedNumber

}`

All other binding related code can go in there too

File: app/src/main/java/com/example/pokedex/pokemonDetails/PokemonDetailsFragment.kt [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    private lateinit var numberTv: TextView
+    private lateinit var detailsViewModel: DetailsViewModel
+    private var pokemonName: String = ""
+    private var pokemonNumber: Int = 0
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        detailsViewModel = ViewModelProvider(this)[DetailsViewModel::class.java]
```
Comment (IfyUnoka4) [inline app/src/main/java/com/example/pokedex/pokemonDetails/PokemonDetailsFragment.kt]: dont work in `onCreateView` move all to `onViewCreated`

File: app/src/main/java/com/example/pokedex/pokemonDetails/detailsModel/Yellow.kt [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package com.example.pokedex.pokemonDetails.detailsModel
+
+data class Yellow(
+    val back_default: String,
+    val back_gray: String,
+    val back_transparent: String,
+    val front_default: String,
+    val front_gray: String,
+    val front_transparent: String
+)
```
Comment (IfyUnoka4) [inline app/src/main/java/com/example/pokedex/pokemonDetails/detailsModel/Yellow.kt]: Remove all unused data classes

File: app/src/main/java/com/example/pokedex/pokemonDetails/pagerFragments/MovesFragment.kt [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        })
+    }
+
+    companion object {
+        @JvmStatic
+        fun newInstance(pokemonName: String) =
+            MovesFragment().apply {
+                arguments = Bundle().apply {
+                    putString("pokemonName", pokemonName)
+
```
Comment (IfyUnoka4) [inline app/src/main/java/com/example/pokedex/pokemonDetails/pagerFragments/MovesFragment.kt]: Why are you passing data through fragment instance??

File: app/src/main/java/com/example/pokedex/pokemonDetails/pagerFragments/StatsFragment.kt [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    private var pokemonName: String? = null
+    private lateinit var statsTV: TextView
+    private lateinit var detailsViewModel: DetailsViewModel
+    private lateinit var statsListView: RecyclerView
+    private lateinit var statsAdapter: StatsAdapter
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
+            pokemonName = it.getString(ARG_PARAM1)
+        }
```
Comment (IfyUnoka4) [inline app/src/main/java/com/example/pokedex/pokemonDetails/pagerFragments/StatsFragment.kt]: use Safeargs to pass data to fragments

File: app/src/main/java/com/example/pokedex/pokemonDetails/viewmodel/DetailsViewModel.kt [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+            try {
+                val response = repository.getPokemonDetails(name)
+                if(response.isSuccessful) {
+                    if (response.body() != null) {
+                        _details.value = response.body()
+                    }
+                }
+            } catch (e: Exception) {
+                Log.d("CHECK_DETAILS", "${e.message}")
+            }
```
Comment (IfyUnoka4) [inline app/src/main/java/com/example/pokedex/pokemonDetails/viewmodel/DetailsViewModel.kt]: remove logs

File: app/src/main/java/com/example/pokedex/pokemonDirectory/PokemonDirectoryFragment.kt [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+
+class PokemonDirectoryFragment : Fragment(), PokedexAdapter.OnItemClickListener {
+    private var _binding: FragmentPokemonDirectoryBinding? = null
+    private val binding get() = _binding!!
+    private lateinit var pokemonViewModel: PokemonViewModel
+    private lateinit var rvDirectory: RecyclerView
+    private lateinit var pgBar: ProgressBar
+    private lateinit var tvInternet: TextView
+    private lateinit var ivInternet: ImageView
+    private lateinit var rvAdapter: PokedexAdapter
```
Comment (IfyUnoka4) [inline app/src/main/java/com/example/pokedex/pokemonDirectory/PokemonDirectoryFragment.kt:21]: Refer to other comments and fix this

File: app/src/main/java/com/example/pokedex/pokemonDirectory/viewmodel/PokemonViewModel.kt [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+            try {
+                val response = repository.getPokemonResult()
+                if (response.isSuccessful) {
+                    if (response.body() != null) {
+                        val pokemonList = response.body()
+                        _pokemon.value = pokemonList?.results ?: listOf()
+                    } else {
+                        Log.d(
+                            "CHECK_LIST",
+                            "Network Call failed with message: ${response.message()}"
```
Comment (IfyUnoka4) [inline app/src/main/java/com/example/pokedex/pokemonDirectory/viewmodel/PokemonViewModel.kt]: Remove logs

File: app/src/main/java/com/example/pokedex/api/ApiService.kt (hunk 1 above)
Comment (Obehi234) [reply app/src/main/java/com/example/pokedex/api/ApiService.kt]: Done

File: app/src/main/java/com/example/pokedex/api/RetrofitInstance.kt (hunk 2 above)
Comment (Obehi234) [reply app/src/main/java/com/example/pokedex/api/RetrofitInstance.kt]: Ok. Created Utils package and object to store base url

BaseURL

File: app/src/main/java/com/example/pokedex/pokemonDetails/pagerFragments/MovesFragment.kt (hunk 7 above)
Comment (Obehi234) [reply app/src/main/java/com/example/pokedex/pokemonDetails/pagerFragments/MovesFragment.kt]: This is for the fragments in my viewpager. It seemed like a more straightforward approach to pass single data. compared to using a nav graph. I needed to pass the pokemon name to be able to correctly display the correct details in the host fragment in the tab layout.

https://stackoverflow.com/questions/52630267/how-to-implement-a-viewpager-with-bottomnavigationview-using-new-navigation-arch

File: app/src/main/java/com/example/pokedex/pokemonDetails/PokemonDetailsFragment.kt (hunk 5 above)
Comment (Obehi234) [reply app/src/main/java/com/example/pokedex/pokemonDetails/PokemonDetailsFragment.kt]: Even the binding? How do i inflate the view? I did this instead:

you can have
`binding.apply {
nameTV = pokemon name
numberTV = formattedNumber

}`

All other binding related code can go in there too

File: app/src/main/java/com/example/pokedex/pokemonDetails/PokemonDetailsFragment.kt (hunk 3 above)
Comment (Obehi234) [reply app/src/main/java/com/example/pokedex/pokemonDetails/PokemonDetailsFragment.kt]: ok

File: app/src/main/java/com/example/pokedex/pokemonDetails/viewmodel/DetailsViewModel.kt (hunk 9 above)
Comment (Obehi234) [reply app/src/main/java/com/example/pokedex/pokemonDetails/viewmodel/DetailsViewModel.kt]: ok

File: app/src/main/java/com/example/pokedex/pokemonDirectory/viewmodel/PokemonViewModel.kt (hunk 11 above)
Comment (Obehi234) [reply app/src/main/java/com/example/pokedex/pokemonDirectory/viewmodel/PokemonViewModel.kt]: Done

File: app/src/main/java/com/example/pokedex/pokemonDirectory/PokemonDirectoryFragment.kt (hunk 10 above)
Comment (Obehi234) [reply app/src/main/java/com/example/pokedex/pokemonDirectory/PokemonDirectoryFragment.kt:21]: Done

File: app/src/main/java/com/example/pokedex/pokemonDetails/detailsModel/Yellow.kt (hunk 6 above)
Comment (IfyUnoka4) [reply app/src/main/java/com/example/pokedex/pokemonDetails/detailsModel/Yellow.kt]: remember this

File: app/src/main/java/com/example/pokedex/pokemonDetails/pagerFragments/MovesAdapter.kt [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+import com.example.pokedex.databinding.FormsRecyclerItemBinding
+import com.example.pokedex.pokemonDetails.detailsModel.Move
+
+class MovesAdapter(private val movesList: List<Move>) :
+    RecyclerView.Adapter<MovesAdapter.MovesViewHolder>() {
+
+    inner class MovesViewHolder(binding: FormsRecyclerItemBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+
+        private val moveName: TextView = binding.abilityName
```
Comment (IfyUnoka4) [inline app/src/main/java/com/example/pokedex/pokemonDetails/pagerFragments/MovesAdapter.kt]: this is not necessary, you can use the binding directly

Comment (Obehi234) [reply app/src/main/java/com/example/pokedex/pokemonDetails/pagerFragments/MovesAdapter.kt]: Done

**Pull request merged**